File: 3740_Minimum_Distance_Between_Three_Equal_Elements_I/solution.py
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        n, L, r = len(nums), [0] * len(nums), 200
        for i in range(n):
            v, p = nums[i] - 1, i + 1
            pk = L[v]
            o, c = pk & 255, pk >> 8
            L[v] = c | (p << 8)
            if o: r = min(r, (p - o) << 1)
        return -(r == 200) | r

File: 3740_Minimum_Distance_Between_Three_Equal_Elements_I/solution.c
int minimumDistance(int* nums, int numsSize) {
    int L[200] = {0}, r = 200;
    for (int i = 0; i < numsSize; i++) {
        int v = nums[i] - 1, p = i + 1;
        int pk = L[v], o = pk & 255, c = pk >> 8;
        L[v] = c | (p << 8);
        if (o) { int d = (p - o) << 1; if (d < r) r = d; }
    }
    return -(r == 200) | r;
}
